python/metatensor-learn/metatensor/learn/data/collate.py

```diff
@@ -0,0 +1,116 @@
+"""
+Module containing a collate function for use in a Dataloader.
+"""
+from typing import List
+import metatensor
+from metatensor import TensorMap
+
+
+def collate_fn(batch: List[List]) -> List[List]:
```

Comment:
Sorry if I misunderstood, but I had the feeling this point was not addressed in the last discussion because it got intermingled with another point. I am okay if we don't do it, but I had the feeling it was not even mentioned

Another question I have comes from looking at this example https://pytorch.org/tutorials/beginner/data_loading_tutorial.html . In this example they work with a dict dataset, and return the batches in the same dict format. While we do not need to replicate this necessary, it seems to me more intuitive. So in your notebook example would be
```python
batch = next(iter(train_dataloader)) # a dict with keys "descriptor", "target", "auxillary"
in_train_batch, out_train_batch, aux_train_batch = batch.values()
```
So here I get the same format out as I put in the data in the Dataset.

https://github.com/lab-cosmo/metatensor/pull/428#discussion_r1417058611

This would also affect the  also the `__get_item__` function in Dataset. So it would return a dict
```python
data_i = dataset[i] -> a dict with keys "descriptor", "target", "auxillary" and inside there is again a list or another dict

```